controllers/machine_controller.go

```diff
@@ -339,32 +350,25 @@ func (r *MachineReconciler) drainNode(cluster *clusterv1.Cluster, nodeName strin
 
 func (r *MachineReconciler) deleteNode(ctx context.Context, cluster *clusterv1.Cluster, name string) error {
 	logger := r.Log.WithValues("machine", name)
-
-	if cluster == nil {
-		// Try to retrieve the Node from the local cluster, if no Cluster reference is found.
-		var node corev1.Node
-		if err := r.Client.Get(ctx, client.ObjectKey{Name: name}, &node); err != nil {
-			return err
+	// Retrieve the Node from the local cluster
+	c := r.Client
+	if cluster != nil {
```

Comment:
@vincepri I'm thinking for now, we can remove this check, and always use `remote.NewClusterClient()`. In the future, if we implement a way to use the in-cluster client when appropriate, ideally we'd implement that logic inside of `remote.NewClusterClient()`, and we wouldn't have to change the logic anywhere else. WDYT?

Reply:
That sounds good to me, let's open an issue though to track it or use #1666

Reply:
1666 works for me